### DISCUSSION TITLE
Semantic versioning for container images?

### DISCUSSION THREAD

**Discussion by mitchdenny** [datalust/seq-tickets#2184] [Help]
Hello from the .NET Aspire team! As you probably know we have an extension for .NET Aspire in our repo for Seq:

https://github.com/dotnet/aspire/tree/main/src/Aspire.Hosting.Seq

We are currently thinking about how to approach container image versioning. Right now Seq is one of the components in our repository which doesn't seem to follow semantic versioning. Based on what I am seeing you have a date-based tagging scheme.

Are there any plans to support semantic versioning? How do you handle compat?

**Comment by nblumhardt** [comment]
Hi Mitch, thanks for dropping by! :wave:

I think it's safe to assume that the YYYY part of a Seq version is equivalent to SemVer "Major" - although we don't necessarily follow SemVer in the Seq API, the container configuration is generally always 100% backwards compatible. Seq and/or its entrypoint handle migration if container arguments/environment variables for an older version are used.

If we're forced to make a breaking change there in the future (or any nontrivial API breakage), chances are it'll be tied to a YYYY major for visibility and ease of communication.

Does this help?
